mlflow/models/utils.py

```diff
@@ -1696,10 +1696,12 @@ def _validate_and_get_model_code_path(model_code_path: str, temp_dir: str) -> st
     model_code_path = os.path.abspath(model_code_path)
```

Comment:
nit: ^ Can we update the docstring above indicate our finding that it only use temp_dir under specific circumstances?